src/audio/module_adapter/module/generic.c

```diff
@@ -495,17 +495,43 @@ EXPORT_SYMBOL(module_set_configuration);
 
 int module_bind(struct processing_module *mod, struct bind_info *bind_data)
 {
+	int ret;
 	const struct module_interface *const ops = mod->dev->drv->adapter_ops;
 
+	if (bind_data->as_sink) {
+		ret = sink_bind(bind_data->as_sink, mod);
+		if (ret)
+			return ret;
+	}
+
+	if (bind_data->as_source) {
+		ret = source_bind(bind_data->as_source, mod);
+		if (ret)
+			return ret;
+	}
+
 	if (ops->bind)
 		return ops->bind(mod, bind_data);
 	return 0;
 }
 
 int module_unbind(struct processing_module *mod, struct unbind_info *unbind_data)
 {
+	int ret;
 	const struct module_interface *const ops = mod->dev->drv->adapter_ops;
 
+	if (unbind_data->from_sink) {
+		ret = sink_unbind(unbind_data->from_sink);
+		if (ret)
```

Comment:
ditto - do we still need to unbind ```from_source``` if this fails ?

Reply:
as_sink (from_sink) and as_source(from source) were mutually exlusive. Api changed to be more clear